alert/log/search.go

```diff
@@ -0,0 +1,146 @@
+package alertlog
+
+import (
+	"context"
+	"database/sql"
+	"github.com/target/goalert/permission"
+	"github.com/target/goalert/search"
+	"github.com/target/goalert/validation"
+	"github.com/target/goalert/validation/validate"
+	"text/template"
+
+	"github.com/lib/pq"
+	"github.com/pkg/errors"
+)
+
+// SearchOptions contains criteria for filtering alert logs.
+type SearchOptions struct {
+	// FilterAlertIDs restricts the log entries belonging to specific alertIDs only.
+	FilterAlertIDs []int
+
+	// Limit restricts the maximum number of rows returned. Default is 15.
+	Limit int
+
+	After SearchCursor `json:"a,omitempty"`
+}
+
+type SearchCursor struct {
+	ID int `json:"i,omitempty"`
+}
+
+var searchTemplate = template.Must(template.New("search").Parse(`
+	SELECT
+		a.id, 
+		a.alert_id,
+		a.timestamp,
+		a.event,
+		a.message,
+		a.sub_type,
+		a.sub_user_id,
+		u.name,
+		a.sub_integration_key_id,
+		i.name,
+		a.sub_hb_monitor_id,
+		hb.name,
+		a.sub_channel_id,
+		nc.name,
+		a.sub_classifier,
+		a.meta
+	FROM alert_logs a
+	{{if .FilterAlertIDs }}
+		LEFT JOIN alerts ON alerts.id = a.alert_id
+	{{end}}
+	LEFT JOIN users u ON u.id = a.sub_user_id
+	LEFT JOIN integration_keys i ON i.id = a.sub_integration_key_id
+	LEFT JOIN heartbeat_monitors hb ON hb.id = a.sub_hb_monitor_id 
+	LEFT JOIN notification_channels nc ON nc.id = a.sub_channel_id
+	WHERE TRUE
+	{{if .FilterAlertIDs}}
```

Comment:
nit: we should use consistent spacing, add `-` to match the `if` statement below.